es6-shim.js

```diff
@@ -1370,32 +1367,29 @@
     },
 
     sign: function sign(value) {
-      var number = Number(value);
-      if (number === 0) { return number; }
-      if (Number.isNaN(number)) { return number; }
-      return number < 0 ? -1 : 1;
+      var x = Number(value);
+      if (x < 0) {
+        return -1;
+      }
+      if (x > 0) {
+        return 1;
+      }
+      return x;
     },
 
     sinh: function sinh(value) {
       var x = Number(value);
-      if (!globalIsFinite(x) || x === 0) { return x; }
-
       if (Math.abs(x) < 1) {
         return (Math.expm1(x) - Math.expm1(-x)) / 2;
       }
-      return (Math.exp(x - 1) - Math.exp(-x - 1)) * Math.E / 2;
+      return (Math.exp(x - 1) - Math.exp(-x - 1)) * (Math.E / 2);
     },
 
     tanh: function tanh(value) {
       var x = Number(value);
-      if (Number.isNaN(x) || x === 0) { return x; }
-      if (x === Infinity) { return 1; }
-      if (x === -Infinity) { return -1; }
-      var a = Math.expm1(x);
-      var b = Math.expm1(-x);
-      if (a === Infinity) { return 1; }
-      if (b === Infinity) { return -1; }
-      return (a - b) / (Math.exp(x) + Math.exp(-x));
+      var a = Math.expm1(2 * x);
+      var b = Math.expm1(2 * x) + 2;
+      return a === b ? 1 : a / b;
```

Comment:
@ljharb , I cannot understand how explicit checks can help, I want to have number of "if" branches as small as possible. But I have replaced ternary operator with `if (a === Infinity && b === Infinity)` .